core/src/com/unciv/ui/worldscreen/bottombar/BattleTable.kt

```diff
@@ -167,39 +179,104 @@ class BattleTable(val worldScreen: WorldScreen): Table() {
         }
 
-        if (!worldScreen.isPlayersTurn || attackableEnemy == null) {
+        if (!worldScreen.isPlayersTurn || attackableTile == null) {
             attackButton.disable()
             attackButton.label.color = Color.GRAY
         }
 
         else {
             attackButton.onClick {
                 try {
-                    Battle.moveAndAttack(attacker, attackableEnemy)
+                    Battle.moveAndAttack(attacker, attackableTile)
+                    worldScreen.mapHolder.unitActionOverlay?.remove() // the overlay was one of attacking
+                    worldScreen.shouldUpdate = true
+                }
+                catch (ex:Exception){
+                    val battleBugPopup = Popup(worldScreen)
+                    battleBugPopup.addGoodSizedLabel("You've encountered a bug that I've been looking for for a while!").row()
+                    battleBugPopup.addGoodSizedLabel("If you could copy your game data (\"Copy saved game to clipboard\" - ").row()
+                    battleBugPopup.addGoodSizedLabel("  paste into an email to yairm210@hotmail.com)").row()
+                    battleBugPopup.addGoodSizedLabel("It would help me figure out what went wrong, since this isn't supposed to happen!").row()
+                    battleBugPopup.addGoodSizedLabel("If you could tell me which unit was selected and which unit you tried to attack,").row()
+                    battleBugPopup.addGoodSizedLabel("  that would be even better!").row()
+                    battleBugPopup.open()
+                }
+            }
+        }
+
+        add(attackButton).colspan(2)
+
+        pack()
+
+        setPosition(worldScreen.stage.width/2-width/2, 5f)
+    }
+
+    private fun simulateNuke(attacker: MapUnitCombatant, targetTile: TileInfo){
+        clear()
+        defaults().pad(5f)
+
+        val attackerNameWrapper = Table()
+        val attackerLabel = attacker.getName().toLabel()
+        attackerNameWrapper.add(UnitGroup(attacker.unit,25f)).padRight(5f)
+        attackerNameWrapper.add(attackerLabel)
+        add(attackerNameWrapper)
+
+        val defenderNameWrapper = Table()
+        for (tile in targetTile.getTilesInDistance(Battle.NUKE_RADIUS)) {
+            val defender: ICombatant = tryGetDefenderAtTile(tile, true) ?: continue
+            val defenderLabel = Label(defender.getName().tr(), skin)
+            when (defender) {
+                is MapUnitCombatant ->
+                    defenderNameWrapper.add(UnitGroup(defender.unit, 25f)).padRight(5f)
+                is CityCombatant -> {
+                    val nation = defender.city.civInfo.nation
+                    val nationIcon = ImageGetter.getNationIcon(nation.name)
+                    nationIcon.color = nation.getInnerColor()
+                    defenderNameWrapper.add(nationIcon).size(25f).padRight(5f)
+                }
+            }
+            defenderNameWrapper.add(defenderLabel).row()
+        }
+        add(defenderNameWrapper).row()
```

Comment:
This also looks like something common to both simulateBattle and simulateNuke, no?

Reply:
No, this section is different for Battle and Nuke, because the Nuke can have multiple defenders (both units and cities), while the Battle - only one (either unit or city).